lib/rules/require-matching-function-name.js

```diff
@@ -93,56 +124,59 @@ module.exports.prototype = {
                 return true;
             }
         }
-    }
-};
 
-/**
- * Fetching name from a Pattern
- *
- * @param {Pattern} pattern - E.g. left side of AssignmentExpression
- * @returns {String|Boolean} - Resolved name or false (if there is an Expression)
- */
-function _resolvePatternName(pattern) {
-    switch (pattern.type) {
-        case 'Identifier':
-            // prop = ...;
-            return pattern.name;
-        case 'Literal':
-            // obj['prop'] = ...;
-            return pattern.value;
-        case 'MemberExpression':
-            // obj.prop = ...;
-            return _resolvePatternName(pattern.property);
-        default:
-            // Something unhandy like obj['x' + 2] = ...;
-            return false;
-    }
-}
+        /**
+         * Fetching name from a Pattern
+         *
+         * @param {Pattern} pattern - E.g. left side of AssignmentExpression
+         * @returns {String|Boolean} - Resolved name or false (if there is an Expression)
+         */
+        function _resolvePatternName(pattern) {
+            switch (pattern.type) {
+                case 'Identifier':
+                    // prop = ...;
+                    return pattern.name;
+                case 'Literal':
+                    // obj['prop'] = ...;
+                    return pattern.value;
+                case 'MemberExpression':
+                    // obj.prop = ...;
+                    return _resolvePatternName(pattern.property);
+                default:
+                    // Something unhandy like obj['x' + 2] = ...;
+                    return false;
+            }
+        }
 
-function checkForMember(assignment, skip, errors) {
-    var _name = _resolvePatternName(assignment.left);
-    if (_name === false || skip(_name, assignment.right)) {
-        return;
-    }
+        function checkForMember(assignment, skip, errors) {
+            if (_this._exceptModuleExports && _isModuleExports(assignment.left)) {
+                return;
+            }
```

Comment:
Fixed and sorry for my poor coding skill. However, [The line you referenced](https://github.com/jscs-dev/node-jscs/pull/1927/files#diff-105ac0edfe6913192922933858c13d7fR82) is also in a function scope https://github.com/jscs-dev/node-jscs/pull/1927/files#diff-105ac0edfe6913192922933858c13d7fR77 so I just cache the `_exceptModuleExports`.